### DISCUSSION TITLE
"kick" in barkour example

### DISCUSSION THREAD

**Discussion by yuichiro-student** [google/brax#534] [Q&A]
In the following tutorial on the quadruped barkour,
https://github.com/google/brax/blob/main/brax/experimental/barkour/tutorial.ipynb
there is a `kick` in the `step` function as follows.

```
  # kick 
    push_interval = 10 
    kick_theta = jax.random.uniform(kick_noise_2, maxval=2 * jp.pi)
    kick = jp.array([jp.cos(kick_theta), jp.sin(kick_theta)])
    kick *= jp.mod(state.info['step'], push_interval) == 0
    qvel = state.pipeline_state.qvel  # pytype: disable=attribute-error
    qvel = qvel.at[:2].set(kick * self._kick_vel + qvel[:2])
    state = state.tree_replace({'pipeline_state.qvel': qvel})
```

It seems that this `kick` is injecting some noise to the velocity of the state in certain frequency.
Would you tell me the purpose of the `kick`? Is it for exploration during training or for something else? Thank you.

**Comment by btaba** [comment]
Hi @yuichiro-student , this was inserted to make the policy robust for sim2real transfer onto a robot
